cmake/arch.cmake

```diff
@@ -1,10 +1,119 @@
-if (CMAKE_SIZEOF_VOID_P EQUAL 4)
+include(CheckSymbolExists)
+
+# Initialization:
+set(is_x64 0)
+set(is_x86 0)
+set(is_aarch64 0)
+set(is_arm  0)
+set(is_arm8 0)
+set(is_arm7 0)
+set(arm_use_neon 0)
+
+option(TG_OWT_ARCH_ARMV7_USE_NEON "Use NEON SIMD instructions when building for ARMv7" ON)
+
+
+# Check for 64-bit x86 (aka x64):
+check_symbol_exists(__x86_64   "stddef.h" HAVE_X64_DEF1)
+check_symbol_exists(__x86_64__ "stddef.h" HAVE_X64_DEF2)
+check_symbol_exists(__amd64    "stddef.h" HAVE_X64_DEF3)
+check_symbol_exists(_M_X64     "stddef.h" HAVE_X64_DEF4)
+
+if ((HAVE_X64_DEF1 OR HAVE_X64_DEF2) OR (HAVE_X64_DEF3 OR HAVE_X64_DEF4))
+    message(STATUS "Processor architecture is 64-bit x86.")
+    set(is_x64 1)
+else()
+
+
+# Check for 32-bit x86:
+check_symbol_exists(__i386   "stddef.h" HAVE_I386_DEF1)
+check_symbol_exists(__i386__ "stddef.h" HAVE_I386_DEF2)
+check_symbol_exists(_M_IX86  "stddef.h" HAVE_I386_DEF3)
+
+if ((HAVE_I386_DEF1 OR HAVE_I386_DEF2) OR HAVE_I386_DEF3)
+    message(STATUS "Processor architecture is 32-bit x86.")
     set(is_x86 1)
```

Comment:
Since this is not just a bitness check anymore, can you fix https://github.com/desktop-app/tg_owt/blob/ceef372ff87c1b6b9ab925cb30ccd00388f8fe73/cmake/init_target.cmake#L78 to be just "if (is_x86)"?

Reply:
Done :)

Reply:
Thank you!